### PR TITLE
medic-gateway api perf tests

### PR DESCRIPTION
**Pull request by alxndrsn**
Add a second set of e2e configurations for running performance tests on Travis.

Because Travis environment is unpredictable, failing perf tests will not break the build, but will show a broken build job which can be inspected to see if performance degredation requires action.

N.B. an e2e test was deleted on this branch because it was consistently failing.  **`TODO`** do not merge while this e2e test is deleted.  Also some of the unrelated log output was removed on Travis.  **`TODO`** this should be re-instated before merging.

The test code itself is in `tests/protractor/performance/api/controllers/sms-gateway.perf.js`, which shares a lot of code with `tests/protractor/e2e/api/controllers/sms-gateway.spec.js`.  This shared code has been pulled out of the latter file into `tests/protractor/gateway-api.utils.js`.

This PR relates to https://github.com/medic/medic-webapp/issues/4110, and further work will done to make these tests pass - either by improving efficiency of the gateway api endpoint, or by increasing the timeouts.

-----
An example Travis job can be seen at https://travis-ci.org/medic/medic-webapp/builds/356406498

screen shot 2018-03-21 at 18 26 48

Note that the build has passed, but the performance tests have failed.

The relevant failure output:
```
Failures:
1) /sms POST for webapp-terminating message processing should return within a reasonable time
  Message:
    Failed: It took 23.98s to respond to the request.  The endpoint should respond within 500ms.
  Stack:
    Error: Failed: It took 23.98s to respond to the request.  The endpoint should respond within 500ms.
        at api.postMessages.then.response (/home/travis/build/medic/medic-webapp/tests/protractor/performance/api/controllers/sms-gateway.perf.js:25:15)
        at ManagedPromise.invokeCallback_ (/home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:1366:14)
        at TaskQueue.execute_ (/home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:2970:14)
        at TaskQueue.executeNext_ (/home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:2953:27)
        at asyncRun (/home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:2860:25)
        at /home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:676:7
        at <anonymous>
        at process._tickCallback (internal/process/next_tick.js:188:7)
2) /sms POST for webapp-originating message processing should return within a reasonable time
  Message:
    Failed: It took 5.657s to respond to the request.  The endpoint should respond within 500ms.
  Stack:
    Error: Failed: It took 5.657s to respond to the request.  The endpoint should respond within 500ms.
        at api.postStatuses.then.response (/home/travis/build/medic/medic-webapp/tests/protractor/performance/api/controllers/sms-gateway.perf.js:46:19)
        at ManagedPromise.invokeCallback_ (/home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:1366:14)
        at TaskQueue.execute_ (/home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:2970:14)
        at TaskQueue.executeNext_ (/home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:2953:27)
        at asyncRun (/home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:2860:25)
        at /home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:676:7
        at <anonymous>
        at process._tickCallback (internal/process/next_tick.js:188:7)
2 specs, 2 failures
Finished in 40.924 seconds
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,8 +14,9 @@ services:
 
 env:
   matrix:
-  - NODE_VERSION=8
-  - NODE_VERSION=9
+  - NODE_VERSION=8 TEST_SUITE=normal
+  - NODE_VERSION=9 TEST_SUITE=normal
+  - NODE_VERSION="lts/*" TEST_SUITE=performance
```
Comment (SCdF) [inline .travis.yml]: does that mean it runs on all lts builds? Surely just running it on 8 or 9 would be good enough?

File: .travis.yml [hunk 2]
```diff
@@ -24,6 +25,10 @@ env:
   - secure: TR1UN2r3beDtIF+VJpLF2ocTv/uxuOKyVWWhzMLre0ZrrBaIP1sLZV7Z4S/km5M92EfPgGW87BogdWE/R+kTTRiPTCbFB/U/3jFKxEZRXKST66YK5JMwsYqb17UtZtFdqEtO9GGbAVzXwpZfMMoXvlKNrors2W32xBm2uIkOSpI=
   - secure: Q8RH65NClRBryxfvlwHQjeR4wGs+GXeUBRIBd1kspAM7Uv+5K3iP+q/TPCIrTL/OpJ5tyCYfeq5hKIFFwrY3JikNXnyHICOjXPE6zSLUm8E8NHaP/orPNWjze2x4yDSRCifqzr2ZiXVq9sxlZfNbZ9eyJxpPFTYDSpuN9T5UEE0=
 
+matrix:
+  allow_failures:
+    - env: NODE_VERSION="lts/*" TEST_SUITE=performance
```
Comment (SCdF) [inline .travis.yml]: How does `allow_failures` translate to Slack and Github? If the perf tests fail but everything else passes does it look green on both of those platforms?

File: .travis.yml [hunk 3]
```diff
@@ -50,19 +55,13 @@ before_script:
   - npm --prefix api install
   - npm --prefix sentinel install
   - ./node_modules/.bin/webdriver-manager update
-  - ./node_modules/.bin/webdriver-manager start &
+  - ./node_modules/.bin/webdriver-manager start >/dev/null &
```
Comment (SCdF) [inline .travis.yml]: We don't want to commit this, either leave as is or use this as an excuse to push it to a file and upload that file to S3

File: .travis.yml [hunk 4]
```diff
@@ -50,19 +55,13 @@ before_script:
...
   - until nc -z localhost 4444; do sleep 1; done
 
 script:
-  - node --stack_size=10000 `which grunt` ci
-
-after_script:
```
Comment (SCdF) [inline .travis.yml]: why are you deleting this?

File: .travis.yml [hunk 5]
```diff
@@ -78,10 +77,6 @@ after_success:
        fi
      fi
 
-after_failure:
```
Comment (SCdF) [inline .travis.yml]: why are you deleting this?

File: .travis.yml (hunk 1 above)
Comment (alxndrsn) [reply .travis.yml]: no, it means it runs on the latest lts build

File: .travis.yml (hunk 2 above)
Comment (alxndrsn) [reply .travis.yml]: Green everywhere, afiact

File: .travis.yml (hunk 3 above)
Comment (alxndrsn) [reply .travis.yml]: Yes, sorry!  Should be another TODO for this at the top of the PR - I deleted this stuff to make failures clearer while I was developing.

File: .travis.yml (hunk 1 above)
Comment (alxndrsn) [reply .travis.yml]: e.g. 
```
$ nvm install lts/*
v8.10.0 is already installed.
Now using node v8.10.0 (npm v5.6.0)
$
```

**Comment by alxndrsn** [comment]
Merged manually.

**Pull request closed without merging**